File: android/src/com/dontflymoney/viewhelper/DfmLicenseCheckerCallback.java
package com.dontflymoney.viewhelper;

import android.content.Intent;
import android.net.Uri;

import com.dontflymoney.baseactivity.Message;
import com.dontflymoney.baseactivity.SmartActivity;
import com.dontflymoney.view.R;
import com.google.android.vending.licensing.LicenseCheckerCallback;
import com.google.android.vending.licensing.Policy;

public class DfmLicenseCheckerCallback implements LicenseCheckerCallback
{
	SmartActivity activity;
	Message message;
	
	public DfmLicenseCheckerCallback(SmartActivity activity, Message message)
	{
		this.activity = activity;
		this.message = message;
	}
	
    public void allow(int reason)
    {
        if (activity.isFinishing())
        {
            // Don't update UI if Activity is finishing.
            return;
        }

        if (reason == Policy.RETRY)
        {
        	message.alertRetryLicense();
        }
        else
        {
            message.alertError("Allow");
        }
    }

    public void dontAllow(int reason)
    {
        if (activity.isFinishing())
        {
            // Don't update UI if Activity is finishing.
            return;
        }

        message.alertError("dont_allow");
        
        if (reason == Policy.RETRY)
        {
        	message.alertRetryLicense();
        }
        else
        {
        	Intent intent = new Intent(Intent.ACTION_VIEW);
        	intent.setData(Uri.parse("market://details?id=" + activity.getPackageName()));
        	activity.startActivity(intent);
        }
    }

	@Override
	public void applicationError(int errorCode)
	{
        if (activity.isFinishing())
        {
            // Don't update UI if Activity is finishing.
            return;
        }

        String genericMessage = activity.getString(R.string.license_error);
        String specificMessage = String.format(genericMessage, errorCode);
        
        message.alertError(specificMessage);
	}
	
}

File: android/DFM/src/main/kotlin/com/darakeon/user/DfmLicenseCheckerCallback.kt
package com.dontflymoney.viewhelper

import android.content.Intent
import android.net.Uri
import com.dontflymoney.api.Site
import com.dontflymoney.view.R
import com.dontflymoney.view.WelcomeActivity
import com.google.android.vending.licensing.LicenseCheckerCallback
import com.google.android.vending.licensing.Policy

class DfmLicenseCheckerCallback(private val activity: WelcomeActivity) : LicenseCheckerCallback {

    override fun allow(reason: Int) {
        if (activity.isFinishing) {
            // Don't update UI if Activity is finishing.
            return
        }

        if (reason == Policy.RETRY) {
            activity.message.alertRetryLicense()
        } else {
            activity.runOnUiThread { activity.GoToApp() }
        }
    }

    override fun dontAllow(reason: Int) {
        if (activity.isFinishing) {
            // Don't update UI if Activity is finishing.
            return
        }

        if (reason == Policy.RETRY) {
            activity.message.alertRetryLicense()
        } else if (Site.IsLocal()) {
            activity.runOnUiThread { activity.KillThem() }
        } else {
            val intent = Intent(Intent.ACTION_VIEW)
            intent.data = Uri.parse("market://details?id=" + activity.packageName)
            activity.startActivity(intent)
        }
    }

    override fun applicationError(errorCode: Int) {
        if (activity.isFinishing) {
            // Don't update UI if Activity is finishing.
            return
        }

        val genericMessage = activity.getString(R.string.license_error)
        val specificMessage = String.format(genericMessage, errorCode)

        activity.message.alertError(specificMessage)
    }

}
